### PR TITLE
Alerting: Update texts for gm rr in the form

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fixes some wrong texts in the rule form when creating a Grafana-managed recording rule.

**Why do we need this feature?**

Texts are confusing when type is Grafana-managed recording rules.
**Who is this feature for?**

**Special notes for your reviewer:**

https://github.com/user-attachments/assets/103740fa-737b-458c-9dfe-00013c8592fa

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx [hunk 1]
```diff
@@ -139,6 +140,10 @@ export function FolderAndGroup({
 
   const defaultGroupValue = group ? { value: group, label: group } : undefined;
 
+  const evaluationDesc = isGrafanaRecordingRule
+    ? t('alerting.folderAndGroup.evaluation.text.recording', 'Define how often the recording rule is evaluated.')
+    : t('alerting.folderAndGroup.evaluation.text.alerting', 'Define how often the alert rule is evaluated');
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx:145]: (inconsistent fullstops here, one has it and the other doesn't)

File: public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx [hunk 2]
```diff
@@ -98,8 +99,8 @@ export function FolderAndGroup({
 
   const styles = useStyles2(getStyles);
 
-  const folder = watch('folder');
-  const group = watch('group');
+  const [folder, group, type] = watch(['folder', 'group', 'type']);
+  const isGrafanaRecordingRule = type === RuleFormType.grafanaRecording;
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx:103]: We have utils for this, don't we? Could we use `isGrafanaRecordingRuleByType` here/similar where necessary?

File: public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx [hunk 3]
```diff
@@ -380,6 +386,16 @@ function EvaluationGroupCreationModal({
...
 
+  const modalTitle = isGrafanaRecordingRule
+    ? t(
+        'alerting.folderAndGroup.evaluation.modal.text.recording',
+        'Create a new evaluation group to use for this recording rule.'
+      )
+    : t(
+        'alerting.folderAndGroup.evaluation.modal.text.alerting',
+        'Create a new evaluation group to use for this alert rule.'
+      );
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx:397]: (fullstops here as well)

File: public/app/features/alerting/unified/components/rule-editor/labels/LabelsFieldInForm.tsx [hunk 4]
```diff
@@ -13,6 +14,16 @@ interface LabelsFieldInFormProps {
...
   const { watch } = useFormContext<RuleFormValues>();
   const labels = watch('labels');
+  const type = watch('type');
+
+  const isGrafanaRecordingRule = type === RuleFormType.grafanaRecording;
+  const text = isGrafanaRecordingRule
+    ? t('alerting.alertform.labels.recording', ' Add labels to your rule.')
+    : t(
+        'alerting.alertform.labels.alerting',
+        ' Add labels to your rule for searching, silencing, or routing to a notification policy.'
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rule-editor/labels/LabelsFieldInForm.tsx:24]: Are the leading spaces intentional here?

File: public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx [hunk 5]
```diff
@@ -231,15 +231,21 @@ function NeedHelpInfoForConfigureNoDataError() {
...
   const docsLink = 'https://grafana.com/docs/grafana/latest/alerting/fundamentals/alert-rules/rule-evaluation/';
 
   return (
     <Stack direction="row" gap={0.5} alignItems="center">
       <Text variant="bodySmall" color="secondary">
-        <Trans i18nKey="alert-rule-form.evaluation-behaviour.description.text">
-          Define how the alert rule is evaluated.
-        </Trans>
+        {isGrafanaRecordingRule ? (
+          <Trans i18nKey="alert-recording-rule-form.evaluation-behaviour.description.text">
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx:241]: Could we namespace this one under `alerting` as well, possibly?

File: public/app/features/alerting/unified/components/expressions/ExpressionStatusIndicator.tsx [hunk 6]
```diff
@@ -1,18 +1,26 @@
...
   isCondition?: boolean;
   onSetCondition?: () => void;
 }
 
 export const ExpressionStatusIndicator = ({ isCondition, onSetCondition }: AlertConditionProps) => {
   const styles = useStyles2(getStyles);
+  const { watch } = useFormContext<RuleFormValues>();
+  const type = watch('type');
+  const isGrafanaRecordingRule = type === RuleFormType.grafanaRecording;
+  const conditionText = isGrafanaRecordingRule ? 'Recording rule condition' : 'Alert condition';
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/expressions/ExpressionStatusIndicator.tsx:19]: question: Does it make sense to have conditions for recording rules?

File: public/app/features/alerting/unified/components/rule-editor/labels/LabelsFieldInForm.tsx [hunk 7]
```diff
@@ -13,6 +14,16 @@ interface LabelsFieldInFormProps {
 export function LabelsFieldInForm({ onEditClick }: LabelsFieldInFormProps) {
   const { watch } = useFormContext<RuleFormValues>();
   const labels = watch('labels');
+  const type = watch('type');
+
+  const isGrafanaRecordingRule = type === RuleFormType.grafanaRecording;
+  const text = isGrafanaRecordingRule
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/labels/LabelsFieldInForm.tsx:20]: Is this still relevant for cloud rules?

File: public/app/features/alerting/unified/components/expressions/ExpressionStatusIndicator.tsx (hunk 6 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/expressions/ExpressionStatusIndicator.tsx:19]: The condition is used as the output for the metric. We send this field to the BE side:
```
record: {
          Metric: name, 
          From: condition 
        },
```
We could update the word though. Something like 'output for metric'?

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/expressions/ExpressionStatusIndicator.tsx:19]: i will update to `recording rule output`

File: public/app/features/alerting/unified/components/rule-editor/labels/LabelsFieldInForm.tsx (hunk 7 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/labels/LabelsFieldInForm.tsx:20]: this component is only rendered for grafana-managed whatever

File: public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx (hunk 3 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx:397]: good catchhh

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx:397]: where?

File: public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx (hunk 2 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx:103]: 💯

File: public/app/features/alerting/unified/components/rule-editor/labels/LabelsFieldInForm.tsx (hunk 7 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/labels/LabelsFieldInForm.tsx:20]: Updating also for cloud recording rules!

File: public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx (hunk 3 above)
Comment (tomratcliffe) [reply public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx:397]: 🙃 No idea. I must be imagining things! Sorry

File: public/app/features/alerting/unified/components/expressions/ExpressionStatusIndicator.tsx [hunk 8]
```diff
@@ -1,18 +1,26 @@
...
 interface AlertConditionProps {
   isCondition?: boolean;
   onSetCondition?: () => void;
 }
 
 export const ExpressionStatusIndicator = ({ isCondition, onSetCondition }: AlertConditionProps) => {
   const styles = useStyles2(getStyles);
+  const { watch } = useFormContext<RuleFormValues>();
+  const type = watch('type');
+  const isGrafanaRecordingRule = type === RuleFormType.grafanaRecording;
```
[the remark this answers is not in the document]
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/expressions/ExpressionStatusIndicator.tsx]: I updated the PR using the helper functions like `isRecordingRuleByType`

**Review by konrad147 — APPROVED** [review]
LGTM!

File: public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx (hunk 3 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx:397]: hahha

**Review by tomratcliffe — APPROVED** [review]
LGTM as well! Thanks

**Review by brendamuir — APPROVED** [review]

**Pull request merged**